crates/types/src/entities/message.rs

```diff
@@ -37,11 +35,20 @@ pub struct Message {
     pub data: Vec<u8>,
     /// The block height from the parent da layer that originated this message
     pub da_height: DaBlockHeight,
-    /// When the message was spent in the Fuel chain
-    pub fuel_block_spend: Option<BlockHeight>, // TODO: get rid of this
 }
 
-impl Message {
+/// Message send from Da layer to fuel by bridge
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct Message {
+    #[cfg_attr(feature = "serde", serde(flatten))]
+    /// The message content relevant to the blockchain.
+    pub message: CompressedMessage,
+    /// Whether a message has been spent or not
+    pub status: MessageStatus,
```

Comment:
We can use `CoinStatus` here for now. We plan to use it in the future for `MessageCoin`s